src/server/status_server.rs

```diff
@@ -89,11 +106,92 @@ impl StatusServer {
     }
 }
 
+// For handling fail points related requests
+#[cfg(not(feature = "no-fail"))]
+fn handle_fail_points_request(
+    req: Request<Body>,
+) -> Box<dyn Future<Item = Response<Body>, Error = hyper::Error> + Send> {
+    let path = req.uri().path().to_owned();
+    let method = req.method().to_owned();
+    let fail_path = format!("{}/", FAIL_POINTS_REQUEST_PATH);
+    let fail_path_has_sub_path: bool = path.starts_with(&fail_path);
+
+    match (method, fail_path_has_sub_path) {
+        (Method::PUT, true) => Box::new(req.into_body().concat2().map(move |chunk| {
+            let (_, name) = path.split_at(fail_path.len());
+            if name.is_empty() {
+                return Response::builder()
+                    .status(StatusCode::UNPROCESSABLE_ENTITY)
+                    .body(MISSING_NAME.into())
+                    .unwrap();
+            };
+
+            let actions = chunk.iter().cloned().collect::<Vec<u8>>();
```

Comment:
Encoding action directly as the request body doesn't look to be a common way for HTTP applications. Usually it should be encoded according to `application/x-www-form-urlencoded` as `action=xxx`.

Reply:
Does hyper support it? I don't want to (and I am not skilled) implement a http server here :P 
Or if there are some easy changes, let me know what to do.